### PR TITLE
Fix issue #4629: [Bug]: Replace claude-3-5-sonnet-20240620 with claude-3-5-sonnet-20241022

### LINKED ISSUE #4629: [Bug]: Replace claude-3-5-sonnet-20240620 with claude-3-5-sonnet-20241022
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Describe the bug and reproduction steps

In various places, claude-3-5-sonnet-20240620 is listed as a recommended/verified model. We should replace this with claude-3-5-sonnet-20241022.

We should *only* replace the model name, and make no other changes. Writing new tests is not necessary.

The only place we should *not* replace the model names are places where both 20240620 and 20241022 are listed (such as the `CACHE_PROMPT_SUPPORTED_MODELS` list).

### OpenHands Installation

Docker command in README

### OpenHands Version

_No response_

### Operating System

None

### Logs, Errors, Screenshots, and Additional Context

_No response_

### PR DESCRIPTION
**Pull request by openhands-agent**
This pull request fixes #4629.

The issue has been successfully resolved. The PR updates all instances of "claude-3-5-sonnet-20240620" to "claude-3-5-sonnet-20241022" across documentation files (including translations), frontend files, and Python files, while properly preserving locations where both model versions need to be listed together (such as in CACHE_PROMPT_SUPPORTED_MODELS and VERIFIED_ANTHROPIC_MODELS).

Automatic fix generated by [OpenHands](https://github.com/All-Hands-AI/OpenHands/) 🙌

### CODE REVIEW

**Comment by neubig** [comment]
It's a bit weird, I don't know why large sections of documentation ended up getting deleted? It doesn't seem that the agent actually did that...

**Comment by tofarr** [comment]
> It's a bit weird, I don't know why large sections of documentation ended up getting deleted? It doesn't seem that the agent actually did that...

Could this be the result of a merge error? It looks like some of those changes were only added yesterday? https://github.com/All-Hands-AI/OpenHands/blame/main/docs/i18n/fr/docusaurus-plugin-content-docs/current/usage/how-to/headless-mode.md

Maybe we need to look at how the agent does merges

**Review by rbren — APPROVED** [review]

File: frontend/__tests__/utils/organizeModelsAndProviders.test.ts [hunk 1]
```diff
@@ -4,7 +4,7 @@ import { organizeModelsAndProviders } from "../../src/utils/organizeModelsAndPro
 test("organizeModelsAndProviders", () => {
   const models = [
     "azure/ada",
-    "azure/gpt-35-turbo",
+    "azure/gpt-35-turbo",
```
Comment (xingyaoww) [inline frontend/__tests__/utils/organizeModelsAndProviders.test.ts:7]:
```suggestion
    "azure/gpt-35-turbo",
```

**Review by xingyaoww — APPROVED** [review]

File: frontend/__tests__/utils/extractModelAndProvider.test.ts [hunk 2]
```diff
@@ -59,9 +59,9 @@ describe("extractModelAndProvider", () => {
       separator: "/",
     });
 
-    expect(extractModelAndProvider("claude-3-5-sonnet-20240620")).toEqual({
+    expect(extractModelAndProvider("claude-3-5-sonnet-20241022")).toEqual({
       provider: "anthropic",
-      model: "claude-3-5-sonnet-20240620",
+      model: "claude-3-5-sonnet-20241022",
```
Comment (enyst) [inline frontend/__tests__/utils/extractModelAndProvider.test.ts:64]: We want to remove the older Sonnet from the UI?

Comment (xingyaoww) [reply frontend/__tests__/utils/extractModelAndProvider.test.ts:64]: I think this is test case - no?

File: frontend/src/utils/verified-models.ts [hunk 3]
```diff
@@ -1,6 +1,6 @@
 // Here are the list of verified models and providers that we know work well with OpenHands.
 export const VERIFIED_PROVIDERS = ["openai", "azure", "anthropic"];
-export const VERIFIED_MODELS = ["gpt-4o", "claude-3-5-sonnet-20240620"];
+export const VERIFIED_MODELS = ["gpt-4o", "claude-3-5-sonnet-20241022"];
```
Comment (enyst) [inline frontend/src/utils/verified-models.ts:3]: We can probably keep the old Sonnet here, if this is _the list of verified models that we know work well_... it did, didn't it? 🤔

File: frontend/__tests__/utils/extractModelAndProvider.test.ts (hunk 2 above)
Comment (enyst) [reply frontend/__tests__/utils/extractModelAndProvider.test.ts:64]: Ah! You're right!

File: frontend/src/utils/verified-models.ts (hunk 3 above)
Comment (xingyaoww) [inline frontend/src/utils/verified-models.ts:3]:
```suggestion
export const VERIFIED_MODELS = ["gpt-4o", "claude-3-5-sonnet-20240620", "claude-3-5-sonnet-20241022"];
```

Comment (xingyaoww) [reply frontend/src/utils/verified-models.ts:3]: Agree here

Comment (xingyaoww) [reply frontend/src/utils/verified-models.ts:3]: It also make sense to me that we only put the new sonnet as verified model to encourage ppl using it 😓

Comment (enyst) [reply frontend/src/utils/verified-models.ts]: I don't know... We already "recommend" somewhere else, I know we also have in addition a little list where we add models as we verify them to be decent...

Comment (enyst) [reply frontend/src/utils/verified-models.ts]: I mean, these are two different things, at least according to the current documentation. We have:
- one recommended model
- verified models
- the rest - all should work (start, run), but we tell people we don't know how well, and don't recommend them... 

Does that make sense?

**Pull request merged**